js/auth.js

```diff
@@ -0,0 +1,139 @@
+import { API_BASE_URL } from '../js/constants.js';
+/**
+ * Registers a new user.
+ * @async
+ * @function
+ * @param {string} name - The name of the user.
+ * @param {string} email - The email of the user.
+ * @param {string} password - The password of the user.
+ * @throws Will throw an error if registration fails.
+ * @returns {Promise<void>} No return value.
+ */
+export async function registerUser(name, email, password) {
+    try {
+        const response = await fetch(`${API_BASE_URL}/social/auth/register`, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify({
+                name,
+                email,
+                password,
+            }),
+        });
+
+        const errorMessageDiv = document.getElementById('error-message');
+
+        if (!response.ok) {
+            const errorData = await response.json();
+            if (errorData.errors && errorData.errors[0].message === 'Profile already exists') {
```

Comment:
Nice initiative for handling this error and presenting a directive for the user on the next step.